### PR TITLE
Spring 07

### PR DESCRIPTION
**Pull request by engineer027**

### CODE REVIEW

File: src/main/java/com/dev/cinema/annotation/LoginValidator.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+public class LoginValidator implements
+        ConstraintValidator<LoginConstraint, String> {
+
+    @Override
+    public void initialize(LoginConstraint contactNumber) {
+    }
+
+    @Override
+    public boolean isValid(String contactField,
```
Comment (yurwing) [inline src/main/java/com/dev/cinema/annotation/LoginValidator.java]: ckecklist
`Don’t forget to rename parameters in isValid() method.`

**Review by yurwing — APPROVED** [review]

**Review by yurwing — COMMENTED** [review]
also delete class HashUtil

File: src/main/java/com/dev/cinema/annotation/FieldsValueMatch.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Constraint(validatedBy = FieldsValueMatchValidator.class)
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldsValueMatch {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/annotation/FieldsValueMatch.java]: redundant empty line

File: src/main/java/com/dev/cinema/annotation/FieldsValueMatchValidator.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.dev.cinema.annotation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldsValueMatchValidator
+        implements ConstraintValidator<FieldsValueMatch, Object> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/annotation/FieldsValueMatchValidator.java]: same

File: src/main/java/com/dev/cinema/annotation/FieldsValueMatchValidator.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    private String fieldMatch;
+
+    public void initialize(FieldsValueMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    public boolean isValid(Object user,
+                           ConstraintValidatorContext context) {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/annotation/FieldsValueMatchValidator.java]: same

File: src/main/java/com/dev/cinema/annotation/LoginValidator.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.dev.cinema.annotation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class LoginValidator implements
+        ConstraintValidator<LoginConstraint, String> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/annotation/LoginValidator.java]: same

File: src/main/java/com/dev/cinema/annotation/LoginValidator.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    @Override
+    public void initialize(LoginConstraint contactNumber) {
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/annotation/LoginValidator.java]: we can remove this

File: src/main/java/com/dev/cinema/annotation/LoginValidator.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+    @Override
+    public boolean isValid(String login,
+                           ConstraintValidatorContext cxt) {
+        return login != null && (login.length() > 8) && (login.length() < 14);
+    }
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/annotation/LoginValidator.java]: redundant empty line

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 8]
```diff
@@ -1,8 +1,16 @@
 package com.dev.cinema.model.dto;
 
+import com.sun.istack.NotNull;
+import javax.validation.constraints.Min;
+
 public class MovieSessionRequestDto {
+    @NotNull
     private String showTime;
+    @NotNull
+    @Min(1)
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: wrappers are not supported by this annotation

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 9]
```diff
@@ -1,8 +1,16 @@
...
     private Long movieId;
+    @NotNull
+    @Min(1)
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: same

File: src/main/java/com/dev/cinema/model/dto/CinemaHallRequestDto.java [hunk 10]
```diff
@@ -1,7 +1,12 @@
 package com.dev.cinema.model.dto;
 
+import com.sun.istack.NotNull;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/CinemaHallRequestDto.java]:
```suggestion
import javax.validation.constraints.NotNull;
```
same for all requestDtos

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 11]
```diff
@@ -1,16 +1,16 @@
...
 import com.sun.istack.NotNull;
-import javax.validation.constraints.Min;
+import javax.validation.constraints.Size;
 
 public class MovieSessionRequestDto {
     @NotNull
     private String showTime;
     @NotNull
-    @Min(1)
+    @Size(min = 1)
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: please see the javadoc 
https://docs.oracle.com/javaee/6/api/javax/validation/constraints/Size.html

are wrappers supported?

File: src/main/java/com/dev/cinema/model/dto/MovieRequestDto.java [hunk 12]
```diff
@@ -1,7 +1,11 @@
 package com.dev.cinema.model.dto;
 
+import com.sun.istack.NotNull;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieRequestDto.java]: what about this?

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
+package com.dev.cinema.model.dto;
+
+import com.dev.cinema.annotation.FieldsValueMatch;
+import com.dev.cinema.annotation.LoginConstraint;
+import com.sun.istack.NotNull;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 14]
```diff
@@ -1,16 +1,13 @@
 package com.dev.cinema.model.dto;
 
 import com.sun.istack.NotNull;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: ?

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**